src/aliceVision/image/io.cpp

```diff
@@ -82,6 +82,72 @@ EImageColorSpace getImageColorSpace(const std::string& imagePath)
     return EImageColorSpace_stringToEnum(colorSpace);
 }
 
+std::string getImageColorSpace(const OIIO::ImageSpec& oiioSpec, const std::string& defaultColorSpace = "", const std::string& imagePath = "")
+{
+    std::string colorSpaceFromFileName = "";
+    if (!imagePath.empty())
+    {
+        colorSpaceFromFileName = getGlobalColorConfigOCIO().getColorSpaceFromFilepath(imagePath);
+    }
+
+    std::map<std::string, std::string> mapColorSpaces;
+
+    for (const auto& m : oiioSpec.extra_attribs)
+    {
+        const std::string name = m.name().string();
+
+        if (name.find("oiio:ColorSpace") != name.npos)
+        {
+            mapColorSpaces.emplace("oiio:ColorSpace", m.get_string());
+        }
+        else if (name.find("AliceVision:ColorSpace") != name.npos)
+        {
+            mapColorSpaces.emplace("AliceVision:ColorSpace", m.get_string());
+        }
+        else if (name.find(":workPlateColourSpace") != name.npos)
+        {
+            mapColorSpaces.emplace("workPlateColourSpace", m.get_string());
+        }
+        else if (name.find(":originalPlateColourSpace") != name.npos)
+        {
+            mapColorSpaces.emplace("originalPlateColourSpace", m.get_string());
+        }
+    }
+
+    std::string colorSpace = defaultColorSpace;
+
+    if (mapColorSpaces.find("AliceVision:ColorSpace") != mapColorSpaces.end())
+    {
+        colorSpace = mapColorSpaces.at("AliceVision:ColorSpace");
+    }
+    else if (mapColorSpaces.find("workPlateColourSpace") != mapColorSpaces.end())
+    {
+        colorSpace = mapColorSpaces.at("workPlateColourSpace");
+    }
+    else if (!colorSpaceFromFileName.empty())
+    {
+        colorSpace = colorSpaceFromFileName;
+    }
+    else if (mapColorSpaces.find("oiio:ColorSpace") != mapColorSpaces.end())
+    {
+        colorSpace = mapColorSpaces.at("oiio:ColorSpace");
+    }
+    else if (mapColorSpaces.find("originalPlateColourSpace") != mapColorSpaces.end())
```

Comment:
should be just after 'workPlateColourSpace'.

Reply:
I don't think so because the result could be worst than using oiio:ColorSpace which does not refer to something original but to something current.